Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py

```diff
@@ -458,13 +469,63 @@ def splunk_results_command(service):
         demisto.results({"Type": 1, "ContentsFormat": "json", "Contents": json.dumps(res)})
 
 
-def fetch_incidents(service):
-    last_run = demisto.getLastRun() and demisto.getLastRun()['time']
-    search_offset = demisto.getLastRun().get('offset', 0)
+def occurred_to_datetime(incident_ocurred_time):
+    incident_time_without_timezone = incident_ocurred_time.split('.')[0]
+    incident_time_datetime = datetime.strptime(incident_time_without_timezone, SPLUNK_TIME_FORMAT)
+    return incident_time_datetime
 
-    incidents = []
+
+def get_latest_incident_time(incidents):
+    def get_incident_time_datetime(incident):
+        incident_time = incident["occurred"]
+        incident_time_datetime = occurred_to_datetime(incident_time)
+        return incident_time_datetime
+
+    latest_incident = max(incidents, key=get_incident_time_datetime)
+    latest_incident_time = latest_incident["occurred"]
+    return latest_incident_time
+
+
+def get_next_start_time(last_run, fetches_with_same_start_time_count, were_new_incidents_found=True):
+    last_run_datetime = occurred_to_datetime(last_run)
+    if were_new_incidents_found:
+        # Decreasing one minute to avoid missing incidents that were indexed late
+        last_run_datetime = last_run_datetime - timedelta(minutes=1)
+
+    # keep last time max 20 mins before current time, to avoid timeout
+    if fetches_with_same_start_time_count >= 20 and not were_new_incidents_found:  # TODO : make parameter
+        last_run_datetime = last_run_datetime + timedelta(minutes=1)
+
+    next_run_without_miliseconds_and_tz = last_run_datetime.strftime(SPLUNK_TIME_FORMAT)
+    next_run = next_run_without_miliseconds_and_tz
+    return next_run
+
+
+def create_incident_custom_id(incident):
+    incident_raw_data = json.loads(incident['rawJSON'])['_raw']
+    incident_occurred = incident['occurred']
+    incident_custom_id = incident_occurred + incident_raw_data
+    incident_custom_id_hash = hashlib.md5(incident_custom_id).hexdigest()
```

Comment:
```suggestion
    raw_hash=hashlib.md5(incident_raw_data).hexdigest()
    incident_custom_id_hash = '{date}_{hash}'.format(date=incident_occurred, hash=raw_hash) 
```
I think it will be better if id will include redouble time. it will be easier to debug.